src/mongoClusters/utils/getMongoClusterMetadata.ts

```diff
@@ -0,0 +1,107 @@
+    'serverInfo.gitVersion'?: string; // Git version of the MongoDB server (non-sensitive)
+    'serverInfo.opensslVersion'?: string; // OpenSSL version used by the server (non-sensitive)
+    'serverInfo.platform'?: string; // Server platform information (non-sensitive)
+    'serverInfo.storageEngines'?: string; // Storage engine used by the server (non-sensitive)
+    'serverInfo.modules'?: string; // List of modules loaded by the server (non-sensitive)
+    'serverInfo.error'?: string; // Error message if fetching server info fails
+
+    'topology.type'?: string; // Type of topology (e.g., replica set, sharded cluster)
+    'topology.numberOfServers'?: string; // Number of servers
+    'topology.minWireVersion'?: string; // Minimum wire protocol version supported
+    'topology.maxWireVersion'?: string; // Maximum wire protocol version supported
+    'topology.error'?: string; // Error message if fetching topology info fails
+
+    'serverStatus.uptime'?: string; // Server uptime in seconds (non-sensitive)
+    'serverStatus.connections.current'?: string; // Current number of connections (non-sensitive)
+    'serverStatus.connections.available'?: string; // Available connections (non-sensitive)
+    'serverStatus.memory.resident'?: string; // Resident memory usage in MB (non-sensitive)
+    'serverStatus.memory.virtual'?: string; // Virtual memory usage in MB (non-sensitive)
+    'serverStatus.error'?: string; // Error message if fetching server status fails
+
+    'hostInfo.json'?: string; // JSON stringified host information
+    'hostInfo.error'?: string; // Error message if fetching host info fails
+}
+
+/**
+ * Retrieves metadata information about a MongoDB cluster.
+ * This data helps improve diagnostics and user experience.
+ * No internal server addresses or sensitive information are read.
+ *
+ * @param client - The MongoClient instance connected to the MongoDB cluster.
+ * @returns A promise that resolves to an object containing various metadata about the MongoDB cluster.
+ *
+ */
+export async function getMongoClusterMetadata(client: MongoClient): Promise<MongoClusterMetadata> {
+    const result: MongoClusterMetadata = {};
+
+    const adminDb = client.db().admin();
+
+    // Fetch build info (server version, git version, etc.)
+    // This information is non-sensitive and aids in diagnostics.
+    try {
+        const buildInfo = await adminDb.command({ buildInfo: 1 });
+        result['serverInfo.version'] = buildInfo.version;
+        result['serverInfo.gitVersion'] = buildInfo.gitVersion;
+        result['serverInfo.opensslVersion'] = buildInfo.opensslVersion;
+        result['serverInfo.platform'] = buildInfo.platform;
+        result['serverInfo.storageEngines'] = (buildInfo.storageEngines as string[])?.join(';');
+        result['serverInfo.modules'] = (buildInfo.modules as string[])?.join(';');
+    } catch (error) {
+        result['serverInfo.error'] = (error as Error).message;
```

Comment:
TIP: In JS you have never be sure that error is instance of Error.  It might be string, number, object, any type. 
I prefer use the following code taking into consideration that TS is type paranoid 
```
const message = error instanceof Error ? error.message : String(error)
```

Reply:
Thanks! :) I'll make sure to remember. Thanks for the tip, and please keep sharing them, I'm still catching up with JS/TS.